lib/code-ribbon-patch.js

```diff
@@ -46,6 +46,29 @@ class CodeRibbonPatch extends Pane {
 
   initialize() {
     this.getElement().classList.add('cr-patch');
+
+    this.onDidAddItem(thing => {
+      crdebug("Patch got new item: ", thing);
+    });
+
+    // probably not useful:
+    // this.onDidMoveItem(thing => {
+    //   crdebug("Patch item was moved: ", thing);
+    // });
+
+    this.onWillRemoveItem(thing => {
+      crdebug("Patch having item removed: ", thing);
+    });
+
+    this.onDidRemoveItem(thing => {
+      crdebug("Patch item removed: ", thing);
+
+      // check if we need to switch this thing with the other patch's thing
+      if (thing.moved == true && thing.destroyed == false) {
+        // crdebug("Switching thing with other Patch...");
+        // TODO
```

Comment:
what does this mean

Reply:
Dunno yet, will address this in a different PR (probably the didadditem / element override one)